core/controllers/learner_dashboard.py

```diff
@@ -0,0 +1,206 @@
+                'category': collection_summary.category,
+                'objective': collection_summary.objective,
+                'language_code': collection_summary.language_code,
+                'last_updated': utils.get_time_in_millisecs(
+                    collection_summary.collection_model_last_updated),
+                'created_on': utils.get_time_in_millisecs(
+                    collection_summary.collection_model_created_on),
+                'status': collection_summary.status,
+                'node_count': collection_summary.node_count,
+                'community_owned': collection_summary.community_owned,
+                'thumbnail_icon_url': (
+                    utils.get_thumbnail_icon_url_for_category(
+                        collection_summary.category)),
+                'thumbnail_bg_color': utils.get_hex_color_for_category(
+                    collection_summary.category),
+            })
+
+        for collection_summary in incomplete_collection_summaries:
+            incomplete_collection_summary_dicts.append({
+                'id': collection_summary.id,
+                'title': collection_summary.title,
+                'category': collection_summary.category,
+                'objective': collection_summary.objective,
+                'language_code': collection_summary.language_code,
+                'last_updated': utils.get_time_in_millisecs(
+                    collection_summary.collection_model_last_updated),
+                'created_on': utils.get_time_in_millisecs(
+                    collection_summary.collection_model_created_on),
+                'status': collection_summary.status,
+                'node_count': collection_summary.node_count,
+                'community_owned': collection_summary.community_owned,
+                'thumbnail_icon_url': (
+                    utils.get_thumbnail_icon_url_for_category(
+                        collection_summary.category)),
+                'thumbnail_bg_color': utils.get_hex_color_for_category(
+                    collection_summary.category),
+            })
+
+        creators_subscribed_to = (
+            subscription_services.get_all_creators_subscribed_to(self.user_id))
+        creators_settings = user_services.get_users_settings(
+            creators_subscribed_to)
+        subscription_list = []
+
+        for index, creator_settings in enumerate(creators_settings):
+            subscription_summary = {
+                'creator_picture_data_url': (
+                    creator_settings.profile_picture_data_url),
+                'creator_username': creator_settings.username,
+                'creator_impact': (
+                    user_services.get_user_impact_score(
+                        creators_subscribed_to[index]))
+            }
+
+            subscription_list.append(subscription_summary)
+
+        self.values.update({
+            'completed_explorations_list': completed_exp_summary_dicts,
```

Comment:
In order to reduce network latency it's important to try and make fewer calls from the Oppia server to the datastore. A common technique to do this is to fetch multiple entities from the datastore in one RPC.

By going to /_ah/stats on your local server you can also see a "waterfall graph" that should help you see how many calls are being made. See the [AppStats](https://cloud.google.com/appengine/docs/standard/python/tools/appstats) reference for more info.

Does this help? You can also look out for usages of ndb.get_multi() and ndb.put_multi() in the codebase.